### DISCUSSION TITLE
"Z as relative" ordering option does not behave in an intuitive fashion.

### DISCUSSION THREAD

**Discussion by spbond** [godotengine/godot-proposals#8186] [2D]
When I set a z index relative to the parent, I expect the canvas item to be ordered relative to the parent but still observe the parent's ordering relative to anything outside of the parent's tree (as well as any items in the tree which have an absolute z index).  As it stands, the engine appears to simply add the child's z index to the parent's which can result in unexpected behavior like the child being drawn behind an item that the parent is in front of.  Solving this requires changing the z indices of potentially everything in your project, and opens the developer up to obscure bugs if they miss the z index of some unrelated object.  As an aid in creating modularized components consisting of multiple sprites that behave as a single entity, it would be nice to have a mechanism for declaring a rendering order for a child which is relative _only_ to it's parent (and by extension, siblings that are also ordered relative to the parent).

**Comment by HephepTeam** [comment]
Just so you know, I feel you. modular character in an isometric view seems a lot harder now that I face this issue.
